pyproject.toml

```diff
@@ -278,6 +278,8 @@ convention = "google"
 
 [tool.pytest.ini_options]
 filterwarnings = [
+    # Turn all warnings not explicitly filtered below into errors
+    "error",
```

Comment:
This is the line that turns warnings into errors (unless they are specifically filtered out).

Reply:
Also, I'm happy to split this into a separate PR if it makes sense.